tests/test_alphanet_tps_regression.rs

```diff
@@ -0,0 +1,197 @@
+        let timeout = Instant::now() + timeout;
+        let nodes = nodes.to_vec();
+        let submitted_transactions = submitted_transactions.clone();
+
+        thread::spawn(move || {
+            let nonces = vec![0u64; nodes.len()];
+            let nonces = Arc::new(RwLock::new(nonces));
+            while Instant::now() < timeout {
+                {
+                    let nodes = nodes.to_vec();
+                    let nonces = nonces.clone();
+                    let submitted_transactions = submitted_transactions.clone();
+                    thread::spawn(move || send_transaction(nodes, nonces, submitted_transactions));
+                }
+                thread::sleep(tx_delay);
+            }
+        })
+    };
+
+    // Delay between checking the nodes.
+    let check_delay = Duration::from_millis(100);
+    // Collection that stores #num of transactions in a block -> when this block was observed.
+    let observed_transactions = Arc::new(RwLock::new(vec![]));
+
+    // Create thread that observes new blocks and counts new transactions in them.
+    let observer_handler = {
+        let timeout = Instant::now() + timeout;
+        let observed_transactions = observed_transactions.clone();
+        thread::spawn(move || {
+            let mut prev_ind = 0;
+            while Instant::now() < timeout {
+                // Get random node.
+                let node = &nodes[sample_queryable_node(&nodes)];
+                let new_ind = node.read().unwrap().user().get_best_block_index();
+                if new_ind > prev_ind {
+                    let blocks = node
+                        .read()
+                        .unwrap()
+                        .user()
+                        .get_shard_blocks_by_index(GetBlocksByIndexRequest {
+                            start: Some(prev_ind + 1),
+                            limit: Some(new_ind),
+                        })
+                        .unwrap();
+                    for b in &blocks.blocks {
+                        observed_transactions
+                            .write()
+                            .unwrap()
+                            .push((b.body.transactions.len() as u64, Instant::now()));
+                    }
+                    prev_ind = new_ind;
+                }
+                thread::sleep(check_delay);
+            }
+        })
+    };
+    transaction_handler.join().unwrap();
+    observer_handler.join().unwrap();
+
+    let bucket_size = Duration::from_secs(10);
+
+    let bucketed_submitted_xacts = bucketize_tps(&submitted_transactions, bucket_size);
+    let mut bucketed_observed_xacts = bucketize_tps(&observed_transactions, bucket_size);
+
+    let _ = stdout()
+        .write(format!("Submitted transactions tps: {:?}", bucketed_submitted_xacts).as_bytes());
+    let _ = stdout()
+        .write(format!("Observed transactions tps: {:?}", bucketed_observed_xacts).as_bytes());
+    let _ = stdout().flush();
```

Comment:
`let _ =` is probably not necessary.

Reply:
When function returns Result it should be used or else compiler returns a warning.

Reply:
Why not unwrap here?

Reply:
Because it can return error.